satpy/readers/mwr_l1b.py

```diff
@@ -0,0 +1,201 @@
+
+    @property
+    def orbit_start(self):
+        """Get the orbit number for the start of data."""
+        return int(self["/attr/orbit_start"])
+
+    @property
+    def orbit_end(self):
+        """Get the orbit number for the end of data."""
+        return int(self["/attr/orbit_end"])
+
+    def get_dataset(self, dataset_id, dataset_info):
+        """Get the data."""
+        raise NotImplementedError("This is not implemented in the Base class.")
+
+    def _get_channel_data(self, dataset_id, dataset_info):
+        channel_data = self[dataset_info["file_key"]]
+        channel_data.coords["n_channels"] = MWR_CHANNEL_NAMES
+        channel_data = channel_data.rename({"n_fovs": "x", "n_scans": "y"})
+        return channel_data.sel(n_channels=dataset_id["name"]).drop_vars("n_channels")
+
+
+
+class AWS_EPS_Sterna_MWR_L1BFile(AWS_EPS_Sterna_BaseFileHandler):
+    """Class implementing the AWS/EPS-Sterna MWR L1b Filehandler.
+
+    This class implements the ESA Arctic Weather Satellite (AWS) and EPS-Sterna
+    MWR Level-1b NetCDF reader. It is designed to be used through the
+    :class:`~satpy.Scene` class using the :mod:`~satpy.Scene.load` method with
+    the reader ``"mwr_l1b_nc"``.
+
+    """
+    def __init__(self, filename, filename_info, filetype_info, auto_maskandscale=True):
+        """Initialize the handler."""
+        super().__init__(filename, filename_info, filetype_info, auto_maskandscale)
+
+        if filetype_info["file_type"].startswith("eps_sterna"):
+            self._feed_horn_group_name = "n_feedhorns"
+        else:
+            self._feed_horn_group_name = "n_geo_groups"
+
+
+    @property
+    def sub_satellite_longitude_start(self):
+        """Get the longitude of sub-satellite point at start of the product."""
+        return self["status/satellite/subsat_longitude_start"].data.item()
+
+    @property
+    def sub_satellite_latitude_start(self):
+        """Get the latitude of sub-satellite point at start of the product."""
+        return self["status/satellite/subsat_latitude_start"].data.item()
+
+    @property
+    def sub_satellite_longitude_end(self):
+        """Get the longitude of sub-satellite point at end of the product."""
+        return self["status/satellite/subsat_longitude_end"].data.item()
+
+    @property
+    def sub_satellite_latitude_end(self):
+        """Get the latitude of sub-satellite point at end of the product."""
+        return self["status/satellite/subsat_latitude_end"].data.item()
+
+    def get_dataset(self, dataset_id, dataset_info):
+        """Get the data."""
+        if dataset_id["name"] in MWR_CHANNEL_NAMES:
+            data_array = self._get_channel_data(dataset_id, dataset_info)
+        elif dataset_id["name"] in NAVIGATION_DATASET_NAMES:
+            data_array = self._get_navigation_data(dataset_id, dataset_info)
+
+        elif dataset_id["name"] in ["longitude", "latitude"]:
+            data_array = self._get_navigation_data(dataset_id, dataset_info)
```

Comment:
Correct me if I'm wrong but this block could be removed if you did `in NAVIGATION_DATASET_NAMES + ["longitude", "latitude"]` in the above `elif`.